### PR TITLE
Separate evaluation worker

### PR DESCRIPTION
**Pull request by codyjhsieh**
Performance Comparisons:

### CODE REVIEW

File: rl_coach/graph_managers/graph_manager.py [hunk 1]
```diff
@@ -502,9 +504,10 @@ def evaluate(self, steps: PlayingStepsType) -> bool:
             if hasattr(self, 'data_store_params'):
                 data_store = self.get_data_store(self.data_store_params)
                 data_store.save_to_store()
-
+            screen.log_title("success")
```
Comment (zach-nervana) [inline rl_coach/graph_managers/graph_manager.py]: There is already a more verbose message below this line, why add another?

File: rl_coach/graph_managers/graph_manager.py [hunk 2]
```diff
@@ -502,9 +504,10 @@ def evaluate(self, steps: PlayingStepsType) -> bool:
...
             screen.success("Reached required success rate. Exiting.")
             return True
+        screen.log_title("fail")
```
Comment (zach-nervana) [inline rl_coach/graph_managers/graph_manager.py]: this message should provide more context. Perhaps something like "Evaluation success rate not yet met.". I am not sure why diva needed though.

File: rl_coach/graph_managers/graph_manager.py [hunk 3]
```diff
@@ -484,7 +486,7 @@ def evaluate(self, steps: PlayingStepsType) -> bool:
         :return: bool, True if the target reward and target success has been reached
         """
         self.verify_graph_was_created()
-
+        screen.log_title("evaluate started")
```
Comment (zach-nervana) [inline rl_coach/graph_managers/graph_manager.py]: This is a nit, but other log titles have a different style. For example: `screen.log_title("Starting to improve {}".format(self.name))`

File: rl_coach/orchestrators/deploy.py [hunk 4]
```diff
@@ -1,18 +1,3 @@
-#
```
Comment (zach-nervana) [inline rl_coach/orchestrators/deploy.py]: was this removal intentional?

File: rl_coach/orchestrators/kubernetes_orchestrator.py [hunk 5]
```diff
@@ -1,20 +1,3 @@
-#
```
Comment (zach-nervana) [inline rl_coach/orchestrators/kubernetes_orchestrator.py]: was this removal intentional

File: rl_coach/training_worker.py [hunk 6]
```diff
@@ -55,16 +39,15 @@ def training_worker(graph_manager, task_parameters):
 
         if graph_manager.should_train():
             steps += 1
+            # screen.log_title('steps: ' + str(steps))
```
Comment (zach-nervana) [inline rl_coach/training_worker.py]: remove old debug printing

**Comment by zach-nervana** [comment]
Sorry about the comments, i just saw the WIP now ...

**Comment by scttl** [comment]
this should now be updated if you want to take a look and determine whether or not to bring it in @galleibo-intel

**Comment by galleibo-intel** [comment]
Hi @scttl, I am not sure, but I think that this is now colliding with zach's PR #295, which attempts at eliminating communication between workers to be based upon S3 uploaded/downloaded files.

**Comment by scttl** [comment]
Yeah I think you're right @galleibo-intel these will now be somewhat at odds with each other.  If the plan is to bring in Zach's PR I say we go ahead and close this one out now (any maybe look to reimplement this again later).
